src/sql/src/plan/statement/ddl.rs

```diff
@@ -4897,6 +4898,53 @@ pub fn plan_alter_schema_rename(
     }))
 }
 
+pub fn plan_alter_schema_swap<F>(
+    scx: &mut StatementContext,
+    name_a: UnresolvedSchemaName,
+    name_b: UnresolvedSchemaName,
+    gen_temp_suffix: F,
+) -> Result<Plan, PlanError>
+where
+    F: Fn(&dyn Fn(&str) -> bool) -> Result<String, PlanError>,
+{
+    let schema_a = scx.resolve_schema(name_a.clone())?;
+    let schema_b = scx.resolve_schema(name_b.clone())?;
+
+    // Schemas need to be in the same database.
+    let db_spec = if schema_a.database() == schema_b.database() {
+        schema_a.database().clone()
+    } else {
+        sql_bail!("cannot swap schemas that are in different databases");
+    };
+    if matches!(db_spec, ResolvedDatabaseSpecifier::Ambient) {
+        sql_bail!("cannot swap schemas that are in the ambient database");
+    };
+
+    // We cannot swap system schemas.
+    if schema_a.id().is_system() || schema_b.id().is_system() {
+        bail_never_supported!("swaping a system schema".to_string())
```

Comment:
Uncovered

Reply:
The word is "swapping", not "swaping".

Reply:
Fixed the typo! As far as coverage, we bail on system schemas earlier than this because of RBAC, but I included this check to be safe